src/marketplace/ListingsView.js

```diff
@@ -78,6 +83,10 @@ class ListingsView extends React.Component<Props, State> {
 	getMarketSalesForTanks(): void  {
 		getMarketTanks(this.state.userId, sales => {
 				this.setState({itemsForSale: sales.filter(sale => !(allComponents.includes(sale.name)))}); 
+				//Once we have all of the tanks being sold add them to the array that is going to house those tanks
+				for(let i = 0; i < sales.length; i++) {
```

Comment:
When you have a lot of tanks in the marketplace, you will be updating the state a ton. Change this to create an array of Tanks first, and then set the state.